app/vmagent/main.go

```diff
@@ -155,6 +156,31 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 		influxQueryRequests.Inc()
 		fmt.Fprintf(w, `{"results":[{"series":[{"values":[]}]}]}`)
 		return true
+	case "/api/v1/query":
```

Comment:
The exposed `/api/v1/query` handler pretends to support all the [MetricsQL](https://github.com/VictoriaMetrics/VictoriaMetrics/wiki/MetricsQL), which isn't true for `vmagent`. This may confuse users. I think it would be better just using `vm_promscrape_targets` metric exposed at `http://vmagent:8429/metrics` page for monitoring the number of targets. This page may be configured for scraping by `vmagent` or Prometheus.

Another option is to query `count(up)` metric from the remote storage configured in `-remoteWrite.url` command-line flag of `vmagent`.

Reply:
yeah, /api/v1/query is not a good api name, users may get confused

but /metrics maybe not too, its good for metrics monitor, but not good for api call, we are using a ui displaying vmagent exporter number, for safety, getting data from vm storage maybe is not a good choice(because this ui used to manage vm cluster, all state should not deply on vm cluster it self)

so, what do you think? a json resp maybe better for api call, can we update /targets api and make it more eradble for api user? or is /targets has been used for some other envir, what query name is more reasonable if we want to make a api, expoer a json state for  targets?

thanks!

Reply:
There are the following options:

* To query `http://vmagent:8429/metrics` page and parse `vm_promscrape_targets` metric value there.
* To add an optional `format` query arg to `/targets`, so it could generate json output if queried via `/targets?format=json`. Let's leave `/targets` page without `format` query arg as is, since its main purpose is to be inspected by humans, not machines.
* To add `/targets/more` handler, which would export more information than the `/targets` page now exports.

Reply:
ok, we will make some change